src/main/java/ch/uzh/ifi/hase/soprafs24/service/UserService.java

```diff
@@ -50,6 +50,47 @@ public User createUser(User newUser) {
     return newUser;
   }
 
+  // User updating:
+  public User updateUser(User userToUpdate) {
+    ServiceHelpers.checkValidString(userToUpdate.getUsername(), "Username");
+    ServiceHelpers.checkValidString(userToUpdate.getName(), "Name");
+    ServiceHelpers.checkValidString(userToUpdate.getPassword(), "Password");
+
+    User existingUser =
+        userRepository.findById(userToUpdate.getUserId())
+            .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "User not found"));
+
+    existingUser.setName(userToUpdate.getName());
+    existingUser.setUsername(userToUpdate.getUsername());
+    existingUser.setPassword(userToUpdate.getPassword());
+
+    userRepository.save(existingUser);
+    userRepository.flush();
+
+    log.debug("Updated Information for User: {}", existingUser);
+    return existingUser;
+  }
+
+  // User deletion:
+  public void deleteUser(Long userId) {
+    User existingUser = userRepository.findById(userId).orElseThrow(
+        () -> new ResponseStatusException(HttpStatus.NOT_FOUND, "User not found"));
+
+    userRepository.delete(existingUser);
+    userRepository.flush();
+
+    log.debug("Deleted User: {}", existingUser);
+  }
+
+  // `checkInput` method in `UserService` checks the input fields of the user for null values.
+  // If any of them are null, it throws a `BAD_REQUEST` exception
+  public void checkInput(User userInput) {
+    if (userInput.getName() == null || userInput.getUsername() == null
+        || userInput.getPassword() == null) {
+      throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Input fields can't be null");
+    }
```

Comment:
what this?

Reply:
This is a method, which I used to implement my methods in UserController, as it was in the issue 52, yet I changed the implementation, didn't need this method anymore, and forgot to delete it, now it's gone